### DISCUSSION TITLE
In `TransformerConv` what is the index: Tensor, ptr: OptTensor, size_i: Optional[int] parameters in `message` method

### DISCUSSION THREAD

**Discussion by Akhil-CM** [pyg-team/pytorch_geometric#9596] [Q&A]
In [TransformerConv](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/conv/transformer_conv.py), there is the usage of `index`, `ptr`, and `size_i` in the `message` method. But it is not passed in the `propagate` method like so: 
```
out = self.propagate(edge_index, query=query, key=key, value=value,
                             edge_attr=edge_attr)
```

Where does the `message` method get these arguments?

**Comment by EdisonLeeeee** [comment]
It's automatically collected by the class methods from `MessagePassing` according to the arguments provided in `propagate`.
